lib/xenops_utils.ml

```diff
@@ -617,3 +617,40 @@ let chunks size lst =
 let best_effort txt f =
   try f ()
   with e -> info "%s: ignoring exception %s" txt (Printexc.to_string e)
+
+
+let json_length rpc = rpc |> Jsonrpc.to_string |> String.length
```

Comment:
These functions are rather expensive because they construct a short-lived string. How often are they called?

Reply:
At most 512 times during startup (it is used to construct the `char_max_encoded_length` array. After that we use the array and don't call this function anymore.